.github/workflows/windows.yml

```diff
@@ -0,0 +1,93 @@
+name: mingw-w64
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  mingw-w64:
+    name: Build and test
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Git branch
+        uses: actions/checkout@v2
+
+      - name: Cache Msys2
+        id: cache-msys2
+        uses: actions/cache@v1.1.0
+        with:
+          path: msys64
+          key: ${{ runner.os }}-msys64
+
+      - name: Get Msys2
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: |
+          Invoke-WebRequest -Outfile msys2.tar.xz -Uri "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.tar.xz"
+          xz -d msys2.tar.xz
+          tar -xf msys2.tar
+          rm msys2.tar
+
+      - name: Add Msys2 to Path
+        run: |
+          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/mingw64/bin"
+          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/usr/bin"
+
+      - shell: cmd
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: bash -lc 'exit'
+
+      - name: Update Pacman
+        run: pacman -Syy --needed --ask=20 --noconfirm
+
+      - name: Cache pacman
+        id: cache-pacman
+        uses: actions/cache@v1.1.0
+        with:
+          path: msys64/var/cache/pacman/pkg
+          key: ${{ runner.os }}-pacman-${{ hashFiles('msys64/var/lib/pacman/sync/mingw64.db') }}
+          restore-keys: |
+            ${{ runner.os }}-pacman-
```

Comment:
I'm mainly using this in hope that whenever a package is updated and downloaded, github will recache the packages instead of keeping stale ones like travis and appveyor, so this is my attempt at it, I'm still not sure on how `action/cache` works with their keys.

I can remove it since it always seems to hit